libs/dal/src/repositories/job/job.schema.ts

```diff
@@ -143,5 +134,80 @@ jobSchema.virtual('environment', {
   justOne: true,
 });
 
+/*
+ * This index was initially created to optimize:
+ * apps/api/src/app/events/usecases/add-job/add-delay-job.usecase.ts
+ */
+jobSchema.index({
+  transactionId: 1,
+  _subscriberId: 1,
+  _templateId: 1,
+  _environmentId: 1,
+  status: 1,
+  type: 1,
+});
+
+/*
+ * This index was initially created to optimize:
+ * apps/api/src/app/events/usecases/message-matcher/message-matcher.usecase.ts
+ */
+jobSchema.index({
+  transactionId: 1,
+  _subscriberId: 1,
```

Comment:
Following the top comment regarding the above query, I think this can be omitted?

Reply:
I agree, the next index below is a copy of this one.
From our discussion with mongo the order of the fields by the esr rule is the most important. Mongo should be smart enough to use the below index if this one is removed.